### PR TITLE
Add SupportsWildcards() attribute to parameters accepting wildcards

### LINKED ISSUE #751: Some parameters accept wildcards need but don’t have the [SupportsWildcards()] attribute
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

The following parameters accept wildcards need but don’t have the [SupportsWildcards()] attribute
•	Find-PSResource -Repository
•	Find-PSResource -Version
•	Save-PSResource -Repository
•	Update-ModuleManifest -AliasesToExport
•	Update-ModuleManifest -CmdletsToExport
•	Update-ModuleManifest -DscResourcesToExport
•	Update-ModuleManifest -FunctionsToExport
•	Update-ModuleManifest -VariablesToExport
•	Update-PSResource -Repository

### Expected behavior

```console
NA
```

### Actual behavior

```console
NA
```

### Error details

_No response_

### Environment data

```powershell
NA
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by alyssa1303**
# PR Summary

This PR adds `[SupportsWildcards()]` attribute to parameters that accept wildcards but aren't currently having it.

## PR Context

Resolves #751 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: src/code/SavePSResource.cs [hunk 1]
```diff
@@ -47,14 +47,15 @@ public sealed class SavePSResource : PSCmdlet
         public string Version { get; set; }
```
Comment (anamnavi) [inline src/code/SavePSResource.cs:47]: The `Version` parameter also supports wildcards. Running the cmdlet with `-Version "2.2.*"` saves the latest version that satisfies the wildcard. Do we want this to be the expected behavior? If so we should add `[SupportsWildcards]` attribute here. If instead, we want only specific versions to be allowed we should fix this in the code.

File: src/code/UpdatePSResource.cs [hunk 2]
```diff
@@ -58,6 +58,7 @@ public sealed class UpdatePSResource : PSCmdlet
         /// Specifies one or more repository names to update packages from.
```
Comment (anamnavi) [inline src/code/UpdatePSResource.cs:58]: The `-Version` parameter also *takes* wildcards, but do we want this as expected behavior? I tried running `Update-PSResource "folder_test" -Version "4.5.*" -Repository PSGallery` but it updates to version 6.5.0.0 which is the latest version. So it doesn't make sense. If we want wildcard support for Version we should correct this. If we don't want that as expected behavior we should ensure wildcards are not passed in for the Version parameter.

**Review by anamnavi — CHANGES_REQUESTED** [review]
Looking great so far! I left some questions we should consider. Also, Get-PSResource -Version parameter also supports wildcards so if we can add that would be great, thanks!

File: src/code/UpdatePSResource.cs (hunk 2 above)
Comment (alyssa1303) [reply src/code/UpdatePSResource.cs:58]: I agree that the current behavior doesn't make sense, so we should discuss this in our next meeting.

**Review by alerickson — APPROVED** [review]

**Pull request merged**